packages/splits-sdk/src/client/splitV1.ts

```diff
@@ -0,0 +1,1309 @@
+        })[0],
+      ],
+      initiateControlTransfer: [
+        encodeEventTopics({
+          abi: this._splitMainAbi,
+          eventName: 'InitiateControlTransfer',
+        })[0],
+      ],
+      cancelControlTransfer: [
+        encodeEventTopics({
+          abi: this._splitMainAbi,
+          eventName: 'CancelControlTransfer',
+        })[0],
+      ],
+      acceptControlTransfer: [
+        encodeEventTopics({
+          abi: this._splitMainAbi,
+          eventName: 'ControlTransfer',
+        })[0],
+      ],
+      makeSplitImmutable: [
+        encodeEventTopics({
+          abi: this._splitMainAbi,
+          eventName: 'ControlTransfer',
+        })[0],
+      ],
+    }
+
+    this.callData = new SplitV1CallData({
+      chainId,
+      publicClient,
+      ensPublicClient,
+      apiConfig,
+      walletClient,
+      includeEnsNames,
+    })
+    this.estimateGas = new SplitV1GasEstimates({
+      chainId,
+      publicClient,
+      ensPublicClient,
+      apiConfig,
+      walletClient,
+      includeEnsNames,
+    })
+  }
+
+  /*
+  /
+  / SPLIT ACTIONS
+  /
+  */
+  // Write actions
+  async submitCreateSplitTransaction(
+    createSplitArgs: CreateSplitConfig,
+  ): Promise<{
+    txHash: Hash
+  }> {
+    const txHash = await this._createSplitTransaction(createSplitArgs)
+    if (!this._isContractTransaction(txHash))
+      throw new Error('Invalid response')
+
+    return { txHash }
+  }
+
+  async createSplit(createSplitArgs: CreateSplitConfig): Promise<{
+    splitAddress: Address
+    event: Log
+  }> {
+    const { txHash } = await this.submitCreateSplitTransaction(createSplitArgs)
+    const events = await this.getTransactionEvents({
+      txHash,
+      eventTopics: this.eventTopics.createSplit,
+    })
+    const event = events.length > 0 ? events[0] : undefined
+    if (event) {
+      const log = decodeEventLog({
+        abi: this._splitMainAbi,
+        data: event.data,
+        topics: event.topics,
+      })
+      if (log.eventName !== 'CreateSplit') throw new Error()
+      return {
+        splitAddress: log.args.split,
+        event,
+      }
+    }
+
+    throw new TransactionFailedError()
+  }
+
+  async submitUpdateSplitTransaction(
+    updateSplitArgs: UpdateSplitConfig,
+  ): Promise<{
+    txHash: Hash
+  }> {
+    const txHash = await this._updateSplitTransaction(updateSplitArgs)
+    if (!this._isContractTransaction(txHash))
+      throw new Error('Invalid response')
+
+    return { txHash }
+  }
+
+  async updateSplit(updateSplitArgs: UpdateSplitConfig): Promise<{
+    event: Log
+  }> {
+    const { txHash } = await this.submitUpdateSplitTransaction(updateSplitArgs)
+    const events = await this.getTransactionEvents({
+      txHash,
+      eventTopics: this.eventTopics.updateSplit,
+    })
+    const event = events.length > 0 ? events[0] : undefined
+    if (event) return { event }
+
+    throw new TransactionFailedError()
+  }
+
+  async submitDistributeTokenTransaction(
```

Comment:
nit: curious your thoughts on this pattern. had to do this so that the react sdk could be more useful (give you info that your tx was pending, then update when the tx was successful).

but it's kinda ugly to have all these `submit...` functions exposed when they are not intended to be used.

might be something better? haven't really looked into it much. and again, not really related to this pr. but maybe a change to make before publishing a prod version of 4.0.0

Reply:
yeah, I wanted to see if I can change this up a bit because it leads to quite a bit of placeholder code, I think once we have the client out and live we can iterate on this for the public release